smartsim/_core/launcher_/slurm/slurm_launcher.py

```diff
@@ -0,0 +1,334 @@
+        )
+        stats = ((id_, parser.parse_sacct(out, id_)[0]) for id_ in ids)
+        # <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+        to_ss_stat = {
+            "RUNNING": JobStatus.RUNNING,
+            "CONFIGURING": JobStatus.RUNNING,
+            "STAGE_OUT": JobStatus.RUNNING,
+            "COMPLETED": JobStatus.COMPLETED,
+            "DEADLINE": JobStatus.COMPLETED,
+            "TIMEOUT": JobStatus.COMPLETED,
+            "BOOT_FAIL": JobStatus.FAILED,
+            "FAILED": JobStatus.FAILED,
+            "NODE_FAIL": JobStatus.FAILED,
+            "OUT_OF_MEMORY": JobStatus.FAILED,
+            "CANCELLED": JobStatus.CANCELLED,
+            "CANCELLED+": JobStatus.CANCELLED,
+            "REVOKED": JobStatus.CANCELLED,
+            "PENDING": JobStatus.PAUSED,
+            "PREEMPTED": JobStatus.PAUSED,
+            "RESV_DEL_HOLD": JobStatus.PAUSED,
+            "REQUEUE_FED": JobStatus.PAUSED,
+            "REQUEUE_HOLD": JobStatus.PAUSED,
+            "REQUEUED": JobStatus.PAUSED,
+            "RESIZING": JobStatus.PAUSED,
+            "SIGNALING": JobStatus.PAUSED,
+            "SPECIAL_EXIT": JobStatus.PAUSED,
+            "STOPPED": JobStatus.PAUSED,
+            "SUSPENDED": JobStatus.PAUSED,
+        }
+        return {id_: to_ss_stat.get(stat, JobStatus.UNKNOWN) for id_, stat in stats}
+
+    def stop_jobs(self, *ids: LaunchedJobID) -> t.Mapping[LaunchedJobID, JobStatus]:
+        """Take a collection of job ids and kill the corresponding processes
+        started by the slurm launcher.
+
+        :param ids: The ids of the launched jobs to stop.
+        :returns: A mapping of ids to their reported status after attempting to
+            stop them.
+        """
+        slurm_infos = tuple(map(self._get_slurm_info_from_job_id, ids))
+        for info in slurm_infos:
+            self._stop_job(info)
+        return self.get_status(*ids)
+
+    @staticmethod
+    def _stop_job(job_info: _LaunchedJobInfo) -> None:
+        """Given the launch information for a slurm process, attempt to kill
+        that process.
+
+        :param job_info: The info for the job that the launcher should attempt
+            to kill.
+        """
+        step_id = job_info.slurm_id
+        # Check if step_id is part of colon-separated run, this is reflected in
+        # a '+' in the step id, so that the format becomes 12345+1.0.  If we
+        # find it it can mean two things: a MPMD srun command, or a
+        # heterogeneous job.  If it is a MPMD srun, then stop parent step
+        # because sub-steps cannot be stopped singularly.
+        is_sub_step = "+" in step_id
+        is_het_job = os.getenv("SLURM_HET_SIZE") is not None
+        # If it is a heterogeneous job, we can stop them like this. Slurm will
+        # throw an error, but will actually kill steps correctly.
+        if is_sub_step and not is_het_job:
+            step_id_, *_ = step_id.split("+", maxsplit=1)
```

Comment:
looks like we're doing parsing outside the parser. consider that the split operation could also be handled in the parser to avoid having tightly coupled the parser to the stop_job implementation. ditto for line 298 above